### DISCUSSION TITLE
Permutations on Generic Objects

### DISCUSSION THREAD

**Discussion by laurinpaech** [morelinq/MoreLINQ#977] [Q&A] [answered]
Hi! I was trying to loop over the permutations of a list of objects. The objects are of a custom type, not builtin. It does work but only  generates a single permutation and therefore iterates once and then stops. Do they need to implement some function for it to work? I couldn't find anything in the documentation. The same code works when using ints

**Comment by viceroypenguin** [comment]
Would you please share an example that demonstrates the problem? I attempted to replicate with custom objects and behavior matched for both `int` and the custom object.

**Comment by atifaziz** [comment]
Works fine with objects of custom types, even anonymous types as shown below:

```c#
using MoreLinq;

var objs = new[]
{
    new { Foo = 1 },
    new { Foo = 2 },
    new { Foo = 3 },
};

Console.WriteLine(objs.Permutations()
                      .Select(e => e.ToDelimitedString(", "))
                      .ToDelimitedString(Environment.NewLine));
```

> Do they need to implement some function for it to work?

No.

**Reply by laurinpaech** [reply]
I tested it again with a smaller list and then it worked. My original list had a size of 59 which would obviously be too large to loop over all permutations. My assumption was that it would still work to test the code before I changed a more realistic input. So I guess due to the size something overflowed or something else happened.

Thanks for the quick replies btw.

**Comment by atifaziz** [comment, marked as answer]
So looks like you indeed spotted a bug. It should be breaking/overflowing instead of returning the wrong result of a single permutation when the source sequence is over some length. In fact, I noticed that for sequences of length 13 and above, it returns the wrong number of permutations!

I have filed this as bug #978.

**Reply by laurinpaech** [reply]
Great, glad that my mistake helped to find a bug! Thanks for the fast responses as well.

**Reply by atifaziz** [reply]
FYI, a fix is in the works. See PR #979 and feel free to review.
